File: app/src/main/java/hu/kts/cmetronome/admob/AdViewWrapper.java
package hu.kts.cmetronome.admob;


import android.arch.lifecycle.Lifecycle;
import android.arch.lifecycle.LifecycleObserver;
import android.arch.lifecycle.OnLifecycleEvent;

import com.google.android.gms.ads.AdRequest;
import com.google.android.gms.ads.AdView;


public class AdViewWrapper implements LifecycleObserver {

    private AdView adView;

    public AdViewWrapper(AdView adView) {
        this.adView = adView;
        AdRequest adRequest = AdMobTestDeviceFilteredBuilderFactory.get().build();
        adView.loadAd(adRequest);
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_RESUME)
    public void onResume() {
        adView.resume();
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_PAUSE)
    public void onPause() {
        adView.pause();
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_DESTROY)
    public void onDestroy() {
        adView.destroy();
    }
}

File: app/src/main/java/hu/kts/cmetronome/admob/AdViewWrapper.kt
package hu.kts.cmetronome.admob


import androidx.lifecycle.DefaultLifecycleObserver
import androidx.lifecycle.LifecycleOwner
import com.google.android.gms.ads.AdView


class AdViewWrapper(private val adView: AdView) : DefaultLifecycleObserver {

    init {
        val adRequest = AdMobTestDeviceFilteredBuilderFactory.get().build()
        adView.loadAd(adRequest)
    }

    override fun onResume(owner: LifecycleOwner) {
        adView.resume()
    }

    override fun onPause(owner: LifecycleOwner) {
        adView.pause()
    }

    override fun onDestroy(owner: LifecycleOwner) {
        adView.destroy()
    }
}
